arabot/modules/waifus.py

```diff
@@ -0,0 +1,97 @@
+    "bully": ("{author} wants to bully someone...", "{author} bullies {target}"),
+    "cringe": ("{author} is cringing", "{target} make{s} {author} cringe"),
+    "cry": ("{author} is crying", "{target} make{s} {author} cry"),
+    "cuddle": ("{author} wants to cuddle someone...", "{author} cuddles {target}"),
+    "dance": ("{author} is dancing", "{author} is dancing with {target}"),
+    "glomp": ("{author} wants to glomp someone...", "{author} glomps {target}"),
+    "handhold": ("{author} wants to hold someone's hand...", "{author} holds {target}'s hand"),
+    "happy": ("{author} is happy", "{target} make{s} {author} happy"),
+    "highfive": ("{author} wants to highfive someone...", "{author} gives {target} a highfive"),
+    "hug": ("{author} wants to hug someone...", "{author} hugs {target}"),
+    "kick": ("{author} wants to kick someone...", "{author} kicks {target}"),
+    "kill": ("{author} wants to kill someone...", "{author} kills {target}"),
+    "kiss": ("{author} wants to kiss someone...", "{author} kisses {target}"),
+    "lick": ("{author} wants to lick someone...", "{author} licks {target}"),
+    "nom": ("{author} is eating", "{author} eats {target}"),
+    "pat": ("{author} wants to pat someone...", "{author} pats {target}"),
+    "poke": ("{author} wants to poke someone...", "{author} pokes {target}"),
+    "slap": ("{author} wants to slap someone...", "{author} slaps {target}"),
+    "smile": ("{author} is smiling", "{target} make{s} {author} smile"),
+    "smug": ("( ͡° ͜ʖ ͡°)", "( ͡° ͜ʖ ͡°)"),
+    "wave": ("{author} waves at someone...", "{author} waves at {author}"),
+    "wink": ("{author} is winking", "{author} winks at {target}"),
+    "yeet": ("{author} wants to yeet someone...", "{author} yeets {target}"),
+}
+
+
+class WaifuCommandsMeta(CogMeta):
+    def __new__(mcls, name, bases, attrs, *args, **kwargs):
+        command_callback = attrs[f"_{name}__callback"]
+        for reaction_type in set(chain(*ImageCategories.values())):
+            attrs[reaction_type] = Command(
+                command_callback,
+                name=reaction_type,
+                usage="[members...]" if reaction_type in REACTION_MAPPING else "",
+            )
+        return super().__new__(mcls, name, bases, attrs, *args, **kwargs)
+
+
+class Waifus(Cog, category=Category.WAIFUS, metaclass=WaifuCommandsMeta):
+    def __init__(self, waifu_client: WaifuAioClient):
+        self.wclient = waifu_client
+
+    async def _get_category_image(self, reaction: str, is_nsfw_channel: bool) -> str:
+        if all(reaction in category for category in ImageCategories.values()):
+            if is_nsfw_channel:
+                category = random.choice((self.wclient.sfw, self.wclient.nsfw))
+            else:
+                category = self.wclient.sfw
+        elif reaction in ImageCategories["nsfw"]:
+            category = self.wclient.nsfw
+        else:
+            category = self.wclient.sfw
```

Comment:
Would this be equivalent and more readable?
```py
is_unrestricted = all(reaction in category for category in ImageCategories.values())
if is_unrestricted and is_nsfw_channel:
    category = random.choice((self.wclient.sfw, self.wclient.nsfw))
elif reaction in ImageCategories["nsfw"]:
    category = self.wclient.nsfw
else: category = self.wclient.sfw
```
Optionally, `is_unrestricted` can be inlined, but it feels long or would look wonky with Python's weird parentheses PEP -- personal opinion.

Reply:
Black will reformat it back to the original state.